gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/event/TimingEvent.java

```diff
@@ -84,46 +92,107 @@
   public static final String JOB_START_TIME = "jobStartTime";
   public static final String JOB_END_TIME = "jobEndTime";
 
-  private final String name;
-  private final Long startTime;
+  @Getter
+  private Long startTime;
+  @Getter
+  private Long endTime;
+  @Getter
+  private Long duration;
   private final EventSubmitter submitter;
   private boolean stopped;
 
   public TimingEvent(EventSubmitter submitter, String name) {
+    super(name);
     this.stopped = false;
-    this.name = name;
     this.submitter = submitter;
     this.startTime = System.currentTimeMillis();
   }
 
   /**
    * Stop the timer and submit the event. If the timer was already stopped before, this is a no-op.
+   * @deprecated Use {@link #close()}
    */
+  @Deprecated
```

Comment:
Is there a reason to mark this as deprecated?

Reply:
Because it uses a deprecated submit method

Reply:
And the alternative way is to use `close`, as mentioned in the comment.

Reply:
Isn't it the same method that `close` uses? I get that there is a new method that is synonymous with this, but what benefit does deprecation bring? (besides marking a bunch of places in code as deprecated)